version: "3.8"

services:
  db:
    ports:
      - 5432:5432
    env_file:
      - ./back/.env

  traefik:
    command:
      # - "--log.level=DEBUG"
      - "--api.insecure=true"
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
    ports:
      # The HTTP port
      - "80:80"
      # The Web UI (enabled by --api.insecure=true)
      - "8081:8080"
    # Traefik has to be in all networks in order to resolve propery the containers it is trying to contact

  cache:
    ports:
      - 6379:6379
    volumes:
      - ./cache:/data

  api:
    build:
      target: dev
    env_file:
      - ./back/.env
    volumes:
      - ./back/.env:/app/back/.env
      - ./back/locales:/app/back/locales
      - ./back/src:/app/back/src
      - ./back/test:/app/back/test
      - ./back/prisma:/app/back/prisma
      - ./packages/core:/app/packages/core
      - ./back/uploads:/app/back/uploads

  front:
    build:
      target: dev
    volumes:
      - ./front/.env:/app/front/.env
      - ./front:/app/front
      - ./packages:/app/packages
